src/cmd/troubleshoot/image_helper.go

```diff
@@ -0,0 +1,125 @@
+package troubleshoot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	removeSchemaRegex      = regexp.MustCompile("^.*//(.*)$")
+	removeApiEndpointRegex = regexp.MustCompile("^(.*)/[^/]*$")
+	registryRegex          = regexp.MustCompile(`^(.*)/linux.*$`)
+	imageRegex             = regexp.MustCompile(`^.*/(linux.*)$`)
+)
+
+type imageInfo struct {
+	registry string
+	image    string
+	version  string
+}
+
+func splitImageName(imageName string) (imageInfo, error) {
+	// some image path examples that work with this function
+	//   aaa00000.dynatrace.com/linux/oneagent
+	//   aaa00000.dynatrace.com/linux/activegate:1.123
```

Comment:
Comment could be a doc-string

```suggestion
// splitImageName creates an instance of imageInfo from an image uri.
// image uris can be in the form of
// * aaa00000.dynatrace.com/linux/oneagent
// * aaa00000.dynatrace.com/linux/activegate:1.123
func splitImageName(imageName string) (imageInfo, error) {
```